elixir/lib/page_iterator.ex

```diff
@@ -0,0 +1,12 @@
+defmodule GithubRepoCloner.PageIterator do
```

Comment:
Rules of the page iterator:

1. Accept an anonymous function and an arguments map that will be passed to the function.
2. The arguments map must include a `page`.
3. Recursively call the function while it returns an `ok` tuple, increasing the page number on each call.
4. Stop recursion whenever an `error` tuple is returned.
5. Page iterator returns the last successful page reached in an `ok` tuple at the end.

Reply:
Tt the very least leave a comment describing the base condition that terminates the repeat iterator.